### ISSUE TITLE
Browserstack with the Jasmine framework prints `undefined` for test names

### ISSUE THREAD

**Issue by stezu** [webdriverio/webdriverio#5102]
**Environment (please complete the following information):**
 - **WebdriverIO version:** 5.18.7
 - **Mode:** WDIO Testrunner
 - **If WDIO Testrunner, running sync/async:** sync
 - **Node.js version:** 12.14.0
 - **NPM version:** 6.13.4
 - **Browser name and version:** Chrome 80
 - **Platform name and version:** Windows 10
 - **Additional wdio packages used (if applicable):** `@wdio/browserstack-service`, `@wdio/jasmine-framework`, `@wdio/junit-reporter`, `@wdio/local-runner`, `@wdio/spec-reporter`, `@wdio/sync`

**Config of WebdriverIO**
```js
{
  runner: 'local',
  baseUrl: 'REDACTED',
  outputDir: 'REDACTED',
  bail: 0,
  framework: 'jasmine',
  specFileRetries: 0,
  reporters: [ 'spec' ],
  jasmineNodeOpts: { defaultTimeoutInterval: 3600000 },
  waitforTimeout: 30000,
  maxInstances: 8,
  logLevel: 'info',
  specs: [
    '!(node_modules)/**/__ui-tests__/**/*.[jt]s?(x)',
    '!(node_modules)/**/?(*.)+(ui|smoke|functional|regression).[jt]s?(x)'
  ],
  automationProtocol: 'webdriver',
  user: 'REDACTED',
  key: 'REDACTED',
  browserstackLocal: true,
  browserstackOpts: {},
  services: [ 'browserstack' ],
  capabilities: [
    {
      project: 'REDACTED',
      build: 'non-ci-build',
      name: 'REDACTED',
      resolution: '2048x1536',
      'browserstack.debug': true,
      'browserstack.console': 'verbose',
      'browserstack.local': true,
      'browserstack.networkLogs': true,
      'browserstack.seleniumLogs': true,
      'browserstack.sendKeys': true,
      'browserstack.video': true,
      os: 'Windows',
      os_version: '10',
      browser: 'chrome',
      browser_version: '80.0',
      browserName: 'chrome',
      browserVersion: '80.0'
    }
}
```

**Describe the bug**
![Screen Shot 2020-03-16 at 11 41 55 AM](https://user-images.githubusercontent.com/3180534/76775039-2b1a3e80-677b-11ea-87c0-8b5e5f0840dc.png)

When using the Jasmine framework, each test name in the Browserstack Automate UI begins with `undefined -`.

**To Reproduce**
Using the configuration above with valid Browserstack credentials and any example test should reproduce the issue.

**Expected behavior**
The `undefined` variable that's being printed in the Browserstack UI _should_ be replaced with the context for the test (in other words, the parent `describe`).

**Log**
N/A

**Additional context**

TL;DR: I believe the bug was introduced in #4354 [here](https://github.com/webdriverio/webdriverio/pull/4354/files#diff-c2ca8ed9f67ba29fa646b301cc338c66L112-L113).

The name reported to Browserstack is assembled within the service in the [`afterTest` hook](https://github.com/webdriverio/webdriverio/blob/d5e86c96a4e0238bd1290b7df0194ef5660ac615/packages/wdio-browserstack-service/src/service.js#L49). The portion that is `undefined` in the Browserstack UI is represented by `test.parent` in that hook.

`parent` would have been conditionally set by the Jasmine service when called after tests (the only case where `id` starts with `spec`), but that plumbing was [removed here](https://github.com/webdriverio/webdriverio/pull/4354/files#diff-c2ca8ed9f67ba29fa646b301cc338c66L112-L113). Therefore, `parent` is never actually set and Browserstack can't use it.

In my mind, there are two decent solutions:
1. Restore the `afterTest` hook so `parent` is set properly, OR
2. Configure the Browserstack service to rely on the `fullName` property which (as far as I can tell) is always correct.

**Comment by christian-bromann** [comment]
I would go with the second solution. PRs are welcome!

**Comment by wswebcreation** [comment]
@christian-bromann 

I'm seeing the same thing in the Sauce Labs service, looks like something broke in the generic reporter

**Comment by christian-bromann** [comment]
Fixed in #5103

**Comment by sdecresc90** [comment]
> **Environment (please complete the following information):**
> 
> * **WebdriverIO version:** 5.18.7
> * **Mode:** WDIO Testrunner
> * **If WDIO Testrunner, running sync/async:** sync
> * **Node.js version:** 12.14.0
> * **NPM version:** 6.13.4
> * **Browser name and version:** Chrome 80
> * **Platform name and version:** Windows 10
> * **Additional wdio packages used (if applicable):** `@wdio/browserstack-service`, `@wdio/jasmine-framework`, `@wdio/junit-reporter`, `@wdio/local-runner`, `@wdio/spec-reporter`, `@wdio/sync`
> 
> **Config of WebdriverIO**
> 
> {
>   runner: 'local',
>   baseUrl: 'REDACTED',
>   outputDir: 'REDACTED',
>   bail: 0,
>   framework: 'jasmine',
>   specFileRetries: 0,
>   reporters: [ 'spec' ],
>   jasmineNodeOpts: { defaultTimeoutInterval: 3600000 },
>   waitforTimeout: 30000,
>   maxInstances: 8,
>   logLevel: 'info',
>   specs: [
>     '!(node_modules)/**/__ui-tests__/**/*.[jt]s?(x)',
>     '!(node_modules)/**/?(*.)+(ui|smoke|functional|regression).[jt]s?(x)'
>   ],
>   automationProtocol: 'webdriver',
>   user: 'REDACTED',
>   key: 'REDACTED',
>   browserstackLocal: true,
>   browserstackOpts: {},
>   services: [ 'browserstack' ],
>   capabilities: [
>     {
>       project: 'REDACTED',
>       build: 'non-ci-build',
>       name: 'REDACTED',
>       resolution: '2048x1536',
>       'browserstack.debug': true,
>       'browserstack.console': 'verbose',
>       'browserstack.local': true,
>       'browserstack.networkLogs': true,
>       'browserstack.seleniumLogs': true,
>       'browserstack.sendKeys': true,
>       'browserstack.video': true,
>       os: 'Windows',
>       os_version: '10',
>       browser: 'chrome',
>       browser_version: '80.0',
>       browserName: 'chrome',
>       browserVersion: '80.0'
>     }
> }
> **Describe the bug** ![Screen Shot 2020-03-16 at 11 41 55 AM](https://user-images.githubusercontent.com/3180534/76775039-2b1a3e80-677b-11ea-87c0-8b5e5f0840dc.png)
> 
> When using the Jasmine framework, each test name in the Browserstack Automate UI begins with `undefined -`.
> 
> **To Reproduce** Using the configuration above with valid Browserstack credentials and any example test should reproduce the issue.
> 
> **Expected behavior** The `undefined` variable that's being printed in the Browserstack UI _should_ be replaced with the context for the test (in other words, the parent `describe`).
> 
> **Log** N/A
> 
> **Additional context**
> 
> TL;DR: I believe the bug was introduced in [#4354](https://github.com/webdriverio/webdriverio/pull/4354) [here](https://github.com/webdriverio/webdriverio/pull/4354/files#diff-c2ca8ed9f67ba29fa646b301cc338c66L112-L113).
> 
> The name reported to Browserstack is assembled within the service in the [`afterTest` hook](https://github.com/webdriverio/webdriverio/blob/d5e86c96a4e0238bd1290b7df0194ef5660ac615/packages/wdio-browserstack-service/src/service.js#L49). The portion that is `undefined` in the Browserstack UI is represented by `test.parent` in that hook.
> 
> `parent` would have been conditionally set by the Jasmine service when called after tests (the only case where `id` starts with `spec`), but that plumbing was [removed here](https://github.com/webdriverio/webdriverio/pull/4354/files#diff-c2ca8ed9f67ba29fa646b301cc338c66L112-L113). Therefore, `parent` is never actually set and Browserstack can't use it.
> 
> In my mind, there are two decent solutions:
> 
> 1. Restore the `afterTest` hook so `parent` is set properly, OR
> 2. Configure the Browserstack service to rely on the `fullName` property which (as far as I can tell) is always correct.

Could it be possible that this issue is present againwhen a test is run with GitHub action with webdriverio 9.25.0,  browser stack-service 9.15.0 and node 20.19.2?
